### PR TITLE
Add support for keyAltName

### PR DESCRIPTION
**Pull request by rozza**
JAVA-3335

### CODE REVIEW

File: bindings/java/mongocrypt/src/test/java/com/mongodb/crypt/capi/MongoCryptTest.java [hunk 1]
```diff
@@ -154,9 +154,50 @@ public void testExplicitEncryptionDecryption() throws IOException, URISyntaxExce
         mongoCrypt.close();
     }
 
+    @Test
+    public void testExplicitEncryptionDecryptionKeyAltName() throws IOException, URISyntaxException {
```
Comment (rozza) [inline bindings/java/mongocrypt/src/test/java/com/mongodb/crypt/capi/MongoCryptTest.java]: A copy of the previous but had to progress the decryptor context as well.

**Comment by rozza** [comment]
Patch: https://evergreen.mongodb.com/version/5d1b623c3627e0746f41d279

File: bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptException.java [hunk 2]
```diff
@@ -44,9 +44,13 @@ public MongoCryptException(final String msg) {
      * @param status the status
      */
     MongoCryptException(final mongocrypt_status_t status) {
-        super(CAPI.mongocrypt_status_message(status, null).toString());
```
[the remark this answers is not in the document]
Comment (kevinAlbs) [reply bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptException.java]: Hmm, do you recall how this came up? If libmongocrypt is returning false (indicating error) on a function that takes a status and not setting the status's message, that's a bug in libmongocrypt.

File: bindings/java/mongocrypt/build.gradle.kts [hunk 3]
```diff
@@ -42,7 +42,7 @@ repositories {
 }
 
 group = "org.mongodb"
-version = "1.0.0-beta2"
+version = "1.0.0-SNAPSHOT"
```
Comment (kevinAlbs) [inline bindings/java/mongocrypt/build.gradle.kts]: I'm not sure what the convention here is. If this is meant to track the libmongocrypt version, than perhaps this should remain beta2 until we tag libmongocrypt with the beta3 tag?

Comment (rozza) [reply bindings/java/mongocrypt/build.gradle.kts]: I just wanted a SNAPSHOT to be published on the snapshots repo until the Java driver work is done. Once thats done a beta3 sounds good.

File: bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptException.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rozza) [reply bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptException.java]: I can't trigger the error now on master - but I was previously able to fail the `isTrue` check and then that raises and IllegalArgumentException instead.  I'll revert the change.

**Comment by rozza** [comment]
Rebased and merged

**Pull request closed without merging**